crates/oxc_linter/src/rules/eslint/no_throw_literal.rs

```diff
@@ -0,0 +1,197 @@
+use oxc_ast::{
+    ast::{AssignmentOperator, AssignmentTarget, Expression, LogicalOperator},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_throw_literal_diagnostic(span: Span, is_undef: bool) -> OxcDiagnostic {
+    let message =
+        if is_undef { "Do not throw undefined." } else { "Expected an error object to be thrown." };
+
+    OxcDiagnostic::warn("Disallow throwing literals as exceptions")
+        .with_help(message)
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoThrowLiteral;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallows throwing literals or non-Error objects as exceptions.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// It is considered good practice to only throw the Error object itself or an object using
+    /// the Error object as base objects for user-defined exceptions. The fundamental benefit of
+    /// Error objects is that they automatically keep track of where they were built and originated.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// throw "error";
+    ///
+    /// throw 0;
+    ///
+    /// throw undefined;
+    ///
+    /// throw null;
+    ///
+    /// var err = new Error();
+    /// throw "an " + err;
+    /// // err is recast to a string literal
+    ///
+    /// var err = new Error();
+    /// throw `${err}`
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// throw new Error();
+    ///
+    /// throw new Error("error");
+    ///
+    /// var e = new Error("error");
+    /// throw e;
+    ///
+    /// try {
+    ///     throw new Error("error");
+    /// } catch (e) {
+    ///     throw e;
+    /// }
+    /// ```
+    NoThrowLiteral,
+    correctness,
```

Comment:
todo: @camc314 we could add a conditional_suggestion when a string is being thrown